modules/core/src/test/java/org/apache/ignite/internal/processors/security/cache/closure/ContinuousQueryWithTransformerRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.security.cache.closure;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import javax.cache.configuration.Factory;
+import javax.cache.event.CacheEntryEvent;
+import org.apache.ignite.cache.query.ContinuousQueryWithTransformer;
+import org.apache.ignite.cache.query.ScanQuery;
+import org.apache.ignite.lang.IgniteClosure;
+import org.apache.ignite.lang.IgniteRunnable;
+import org.junit.Test;
+
+/**
+ * Testing operation security context when the ScanQuery, RemoteFilterFactory or TransformerFactory of a
+ * ContinuousQueryWithTransformer is executed on remote node.
+ * <p>
+ * The initiator node broadcasts a task to 'run' node that starts a ContinuousQueryWithTransformer's component. That
+ * component is executed on 'check' node. On every step, it is performed verification that operation securitycontext is
+ * the initiator context.
+ */
+public class ContinuousQueryWithTransformerRemoteSecurityContextCheckTest extends
+    AbstractContinuousQueryRemoteSecurityContextCheckTest {
+    /** Test InitialQuery of ContinuousQueryWithTransformer. */
+    @Test
+    public void testInitialQuery() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setInitialQuery(new ScanQuery<>(INITIAL_QUERY_FILTER));
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(false));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test RemoteFilterFactory of ContinuousQueryWithTransformer. */
```

Comment:
Convert to multi-line javadoc according to coding guidelines, change text 'Test' to 'Tests'.